hathor/manager.py

```diff
@@ -344,11 +344,13 @@ def generate_mining_block(self, timestamp: Optional[float] = None,
         """
         from hathor.transaction.scripts import create_output_script
 
-        if not timestamp:
+        original_timestamp = timestamp
+
+        if timestamp is None:
             timestamp = max(self.tx_storage.latest_timestamp, self.reactor.seconds())
 
         if parent_block_hash is None:
-            tip_blocks = self.tx_storage.get_best_block_tips(timestamp)
+            tip_blocks = self.tx_storage.get_best_block_tips(original_timestamp)
```

Comment:
I think it's important to add a comment to explain why we need to use `original_timestamp` (to keep the None and get the cache), so we don't get confused in the future

Reply:
Fixed. Thanks! Can you review the comment please?